File: app/src/main/java/com/example/pizzamaniaapp/app/data/local/AppDatabase.java
package com.example.pizzamaniaapp.app.data.local;

import android.content.Context;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import com.example.pizzamaniaapp.app.data.dao.CartItemDao;
import com.example.pizzamaniaapp.app.data.dao.MenuItemDao;
import com.example.pizzamaniaapp.app.data.entities.CartItem;
import com.example.pizzamaniaapp.app.data.entities.MenuItem;

@Database(entities = {MenuItem.class, CartItem.class}, version = 1, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {
    private static volatile AppDatabase INSTANCE;

    public abstract MenuItemDao menuItemDao();
    public abstract CartItemDao cartItemDao();

    public static AppDatabase get(Context context) {
        if (INSTANCE == null) {
            synchronized (AppDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                                    AppDatabase.class, "pizza_mania.db")
                            .fallbackToDestructiveMigration()
                            .build();
                }
            }
        }
        return INSTANCE;
    }
}

File: app/src/main/java/com/example/pizzamaniaapp/app/data/local/AppDatabase.kt
package com.pizzamania.data.local

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters

/**
 * Room database for PizzaMania.
 *
 * Stores:
 *   - Cart items (CartItemEntity)
 *   - Menu cache entries (MenuCacheEntity)
 *
 * @TypeConverters is used to serialize complex types like List<String> (toppings).
 *
 * ⚠️ IMPORTANT: Increment the version number every time you change entity schemas.
 */
@Database(
    entities = [
        CartItemEntity::class,   // ✅ your cart table entity
        MenuCacheEntity::class   // ✅ your menu cache table
    ],
    version = 4,                 // bumped since toppings was added
    exportSchema = false
)
@TypeConverters(Converters::class)   // used to persist List<String> → CSV
abstract class PizzaDatabase : RoomDatabase() {

    /** DAO for managing Cart items */
    abstract fun cartDao(): CartDao

    /** DAO for managing Menu cache */
    abstract fun menuCacheDao(): MenuCacheDao
}